ptf/saivlan.py

```diff
@@ -0,0 +1,2553 @@
+                remaining_timeout = remaining_timeout - 0.1
+                (rcv_device, rcv_port, rcv_pkt, _) = test.dataplane.poll(
+                    port_number=port, timeout=0.1, filters=get_filters())
+                if rcv_device != device_number:
+                    continue
+                for pkt in pkts:
+                    logging.debug("Checking for pkt on device %d, port %d",
+                                  device_number, port)
+                    if ptf.dataplane.match_exp_pkt(pkt, rcv_pkt):
+                        pkt_cnt += 1
+                        rcv_ports.add(port_list.index(rcv_port))
+                        port_sub_list_failures = []
+                        port_sub_list_poll_success = True
+                        break
+                    else:
+                        port_sub_list_failures.append(
+                            (port, DataPlane.PollFailure(pkt, [rcv_pkt], 1)))
+                if port_sub_list_poll_success and no_flood:
+                    break
+        # Either no expected packets received or unexpected packets recieved
+        if not port_sub_list_poll_success or port_sub_list_failures:
+            port_tuple = tuple(port_list)
+            failures.setdefault(port_tuple, [])
+            failures[port_tuple] = failures[port_tuple] + \
+                port_sub_list_failures
+        rcv_idx.append(rcv_ports)
+
+    verify_no_other_packets(test)
+    if failures:
+        def format_per_port_failure(port, failure):
+            """
+            Format one port
+            Args:
+              port (int): port
+              failure (str): message
+            """
+            return "On port {}\n{}".format(port, failure.format())
+
+        def format_per_port_failures(fail_list):
+            """
+            Format one port all failures
+            Args:
+              fail_list (array): port and failures
+            """
+            return "\n".join([format_per_port_failure(port, failure)
+                              for (port, failure) in fail_list])
+
+        def format_port_list_failures(port_list, fail_list):
+            """
+            Format all port all failures
+            Args:
+              port_list (array): ports
+              fail_list (array): port and failures
+            """
+            return "None of the exp pkts rx'd for port list {}: \n{}".format(
+                port_list, format_per_port_failures(fail_list))
+        failure_report = "\n".join([
+            format_port_list_failures(port_list, fail_list)
+            for port_list, fail_list in list(failures.items())
+        ])
+        test.fail(
+            "Did not receive expected packets on any of {} for device {}. \n{}"
+            .format(ports, device_number, failure_report))
+
+    test.assertTrue(
+        pkt_cnt >= len(ports),
+        "Did not receive pkt on one of ports %r for device %d" %
+        (ports, device_number))
+    return rcv_idx
```

Comment:
This helper part should be removed. You should be able to replace verify_any_packet_on_ports_list() function with verify_packets_on_multiple_port_lists() after refactoring messy code between 1430-1517